overlord/servicestate/quota_handlers.go

```diff
@@ -341,47 +407,53 @@ func quotaUpdateGroupLimits(grp *quota.Group, limits quota.Resources) error {
 	return grp.UpdateQuotaLimits(currentQuotas)
 }
 
-func quotaUpdate(st *state.State, action QuotaControlAction, allGrps map[string]*quota.Group) (*quota.Group, map[string]*quota.Group, error) {
+func quotaUpdate(st *state.State, action QuotaControlAction, allGrps map[string]*quota.Group) (*quota.Group, map[string]*quota.Group, bool, error) {
 	// make sure the group exists
 	grp, ok := allGrps[action.QuotaName]
 	if !ok {
-		return nil, nil, fmt.Errorf("group %q does not exist", action.QuotaName)
+		return nil, nil, false, fmt.Errorf("group %q does not exist", action.QuotaName)
 	}
 
 	// check that ParentName is not set, since we don't currently support
 	// re-parenting
 	if action.ParentName != "" {
-		return nil, nil, fmt.Errorf("group %q cannot be moved to a different parent (re-parenting not yet supported)", action.QuotaName)
+		return nil, nil, false, fmt.Errorf("group %q cannot be moved to a different parent (re-parenting not yet supported)", action.QuotaName)
 	}
 
 	modifiedGrps := []*quota.Group{grp}
 
 	// ensure that the group we are modifying does not contain a mix of snaps and sub-groups
 	// as we no longer support this, and existing quota groups might have this
 	if err := ensureGroupIsNotMixed(action.QuotaName, allGrps); err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
 	// now ensure that all of the snaps mentioned in AddSnaps exist as snaps and
 	// that they aren't already in an existing quota group
 	if err := validateSnapForAddingToGroup(st, action.AddSnaps, action.QuotaName, allGrps); err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
 	// append the snaps list in the group
 	grp.Snaps = append(grp.Snaps, action.AddSnaps...)
 
+	// store the current status of journal quota, if it changes we need
+	// to refresh the profiles for the snaps in the groups
+	refreshProfiles := grp.JournalLimit != nil
+
 	// update resource limits for the group
 	if err := quotaUpdateGroupLimits(grp, action.ResourceLimits); err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
 	// update the quota group state
 	allGrps, err := internal.PatchQuotas(st, modifiedGrps...)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
-	return grp, allGrps, nil
+
+	refreshProfiles = refreshProfiles != (grp.JournalLimit != nil)
```

Comment:
I suggest renaming the previous refreshProfiles variable to `hadJournalLimit` and:
```suggestion
	hasJournalLimit := grp.JournalLimit != nil
	refreshProfiles := hasJournalLimit != hadJournalLimit
```